pkg/controller/daemon/controller.go

```diff
@@ -518,6 +505,21 @@ func (dsc *DaemonSetsController) syncDaemonSet(key string) error {
 	return nil
 }
 
+func nodeShouldRunDaemonPod(node *api.Node, ds *extensions.DaemonSet) bool {
+	// Check if the node satisfies the daemon set's node selector.
+	nodeSelector := labels.Set(ds.Spec.Template.Spec.NodeSelector).AsSelector()
+	shouldRun := nodeSelector.Matches(labels.Set(node.Labels))
+	// If the daemon set specifies a node name, check that it matches with nodeName.
+	nodeName := node.Name
```

Comment:
Is it really required to alias node.Name? Seems to be used in only one place (in the next line).